skbio/metadata/_interval.py

```diff
@@ -0,0 +1,465 @@
+                    num_keys, self.dropped))
+        return summary
+
+    @experimental(as_of='0.4.2-dev')
+    def __str__(self):
+        return self.__repr__()
+
+    @experimental(as_of='0.4.2-dev')
+    def drop(self):
+        self._interval_metadata.drop(intervals=self.intervals,
+                                     boundaries=self.boundaries,
+                                     metadata=self.metadata)
+        self.boundaries = None
+        self.intervals = None
+        self._interval_metadata = None
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def boundaries(self):
+        if self.dropped:
+            raise RuntimeError('Cannot retrieve boundaries from.'
+                               'dropped Interval object.')
+        return self._boundaries
+
+    @boundaries.setter
+    @experimental(as_of='0.4.2-dev')
+    def boundaries(self, value):
+        if value is None:
+            self._boundaries = None
+        else:
+            for boundary in value:
+                _assert_valid_boundary(boundary)
+
+            if self.dropped:
+                raise RuntimeError('Cannot change boundaries to dropped '
+                                   'Interval object.')
+            self._boundaries = value
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def intervals(self):
+        if self.dropped:
+            raise RuntimeError('Cannot retrieve intervals from.'
+                               'dropped Interval object.')
+        return self._intervals
+
+    @intervals.setter
+    @experimental(as_of='0.4.2-dev')
+    def intervals(self, value):
+        if value is None:
+            self._intervals = None
+        else:
+            for interval in value:
+                _assert_valid_interval(interval)
+
+            if self.dropped:
+                raise RuntimeError('Cannot change intervals to dropped '
+                                   'Interval object.')
+            self._intervals = value
+            if self._interval_metadata is not None:
+                self._interval_metadata._is_stale_tree = True
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def metadata(self):
+        if self.dropped:
+            raise RuntimeError('Cannot retrieve intervals from.'
+                               'dropped Interval object.')
+        return self._metadata
+
+    @metadata.setter
+    @experimental(as_of='0.4.2-dev')
+    def metadata(self, value):
+        if self.dropped:
+            raise RuntimeError('Cannot change metadata to dropped '
+                               'Interval object.')
+        if not isinstance(value, dict):
+            raise TypeError("metadata must be a dict, not %r" % value)
+        self._metadata = value
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def dropped(self):
+        return self._interval_metadata is None
+
+
+class IntervalMetadata():
+    """ Stores Interval objects """
+    def __init__(self):
```

Comment:
The public API needs to be documented. Users will not know how to construct this object.